tderenko/php_advanced_hw
# [homework] number 4

# issue
Клас повинен містити три приватні властивості:
```
red
green
blue
```
Конструктор класу повинен приймати параметри для кожного кольору.
Для всіх властивостей створити публічні гетери та приватні сеттери (методи які починаються на слово (get... | set...) та виводять або задають значення полям класу) .
У сеттерах кольорів перевірити значення, що передається на діапазон від 0 до 255. Якщо число, що передається, виходить за діапазон - викинути виняток (Exception).
Реалізувати метод equals, який порівнюватиме об'єкти кольорів і повертатиме true або false.
Реалізуватиме статичний метод random, який повертатиме об'єкт RGB кольору з випадковими значеннями властивостей red, green, blue.
Реалізувати метод mix, який буде кольору. Змішувати (обчислювати середнє для кожної властивості кольору).
Приклад методу mix:
```
$color = new Color(200, 200, 200);
$mixedColor = $color->mix(new Color(100, 100, 100));
$mixedColor->getRed(); // 150
$mixedColor->getGreen(); // 150
$mixedColor->getBlue(); // 150
```

# Test Plan
1. run `docker compose up -d`
2. go to `http:\\localhost:8080`
3. you should see the homework result
![image](https://user-images.githubusercontent.com/18249681/180658196-a3e213a3-bd5e-437d-9c35-43a32a700d7a.png)
